src/Pretzel.Logic/Extensibility/IHaveCommandLineArgs.cs

```diff
@@ -1,10 +1,14 @@
-using NDesk.Options;
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.Composition;
 
 namespace Pretzel.Logic.Extensibility
 {
     public interface IHaveCommandLineArgs
     {
-        void UpdateOptions(OptionSet options);
-        string[] GetArguments(string command);
+        void UpdateOptions(IList<Option> options);
+
+        void BindingCompleted();
```

Comment:
It seems that this is not used, or have I missing something?

Reply:
If you mean the `BindingCompleted` method you are right. It's not used in pretzel it self (cause there are only 2 argument extentions) the purpose of this methods argument validation and path corrections (relative paths to absolutes and so on)

Reply:
It was about `BindingCompleted` thanks, for the explanation.
Regarding the options list, I'd say that the plugins author should just to pass their own options, but never have the opportunity to modify the current list of options.

Reply:
now they need to provide their own collection